### ISSUE TITLE
No error is shown for a wrong type for tuple unpacking

### ISSUE THREAD

**Issue by lancelote** [intellij-rust/intellij-rust#4211]
## Environment

* **IntelliJ Rust plugin version:** 0.2.103.2698-192-nightly
* **Rust toolchain version:** rustc 1.36.0 (a53f9df32 2019-07-03)
* **IDE name and version:** CLion 2019.3 Nightly 193.1846
* **Operating system:** macOS 10.14.4

## Problem description

Consider the sample:

```rust
fn main() {
    let (foo, bar) = baz();
}

fn baz() -> String {
    String::from("hello world")
}
```

Should not the wrong type for tuple unpacking be reported?
